src/library/scala/collection/Searching.scala

```
@@ -88,7 +88,7 @@ object Searching {
     @tailrec
     private def binarySearch[B >: A](elem: B, from: Int, to: Int)
     (implicit ord: Ordering[B]): SearchResult = {
-      if ((to-from) == 1) InsertionPoint(from) else {
+      if ((to-from) == 1) InsertionPoint(from + 1) else {
```

Shouldn't this be `to` instead of `from+1` since we've already established that they're the same thing?  Or, alternatively, to make this less fragile to bad input (e.g. `to` less than `from`), test `to <= from+1` (but then I don't know whether `to` or `from+1` is more expected--maybe just throw an exception instead)?